core/src/main/scala/org/apache/spark/util/ShutdownHookManager.scala

```diff
@@ -142,6 +142,7 @@ private[spark] object ShutdownHookManager extends Logging {
     addShutdownHook(DEFAULT_SHUTDOWN_PRIORITY)(hook)
   }
 
+  val DEFAULT_HOOK_CALLER_DEPTH = 3
```

Comment:
I think this is kind of arbitrary and the result won't be so helpful. Let's just let callers specify a meaningful string to identify the hook, and then specify that throughout the code.

Reply:
you are right I change it to hook name